src/rpc/blockchain.cpp

```diff
@@ -3109,7 +3110,7 @@ return RPCHelpMan{
         data.pushKV("blocks",                (int)chain.Height());
         data.pushKV("bestblockhash",         tip->GetBlockHash().GetHex());
         data.pushKV("difficulty", GetDifficulty(*tip));
-        data.pushKV("verificationprogress",  GuessVerificationProgress(Params().TxData(), tip));
+        data.pushKV("verificationprogress", validated ? 1 : GuessVerificationProgress(Params().TxData(), tip));
```

Comment:
Note to self, using the "validated" field isn't a good check here, best to use blockheight equals headers as in getblockchaininfo.